scipy/ndimage/tests/test_measurements.py

```diff
@@ -1271,147 +1446,153 @@ def test_watershed_ift03(self):
+        data[...] = np.asarray([[0, 1, 0, 0, 0, 1, 0],
+                                [0, 1, 0, 0, 0, 1, 0],
+                                [0, 1, 0, 0, 0, 1, 0],
+                                [0, 1, 1, 1, 1, 1, 0],
+                                [0, 0, 0, 0, 0, 0, 0],
+                                [0, 0, 0, 0, 0, 0, 0]], dtype=np.uint8)
+        data = xp.asarray(data)
+        markers = xp.asarray([[-1, 0, 0, 0, 0, 0, 0],
+                              [0, 0, 0, 1, 0, 0, 0],
                               [0, 0, 0, 0, 0, 0, 0],
-                              [0, 0, 0, 0, 0, 0, 0]], np.uint8)
-        markers = np.array([[-1, 0, 0, 0, 0, 0, 0],
-                            [0, 0, 0, 1, 0, 0, 0],
-                            [0, 0, 0, 0, 0, 0, 0],
-                            [0, 0, 0, 0, 0, 0, 0],
-                            [0, 0, 0, 0, 0, 0, 0],
-                            [0, 0, 0, 0, 0, 0, 0]], np.int8)
-        out = np.zeros(shape, dtype=np.int16)
-        out = out.transpose()
-        ndimage.watershed_ift(data, markers,
-                              structure=[[1, 1, 1],
-                                         [1, 1, 1],
-                                         [1, 1, 1]],
+                              [0, 0, 0, 0, 0, 0, 0],
+                              [0, 0, 0, 0, 0, 0, 0],
+                              [0, 0, 0, 0, 0, 0, 0]], dtype=xp.int8)
+        out = xp.zeros(shape, dtype=xp.int16)
+        out = out.T
+        structure=xp.asarray([[1, 1, 1],
+                              [1, 1, 1],
+                              [1, 1, 1]])
+        ndimage.watershed_ift(data, markers, structure=structure,
                               output=out)
         expected = [[-1, 1, 1, 1, 1, 1, -1],
                     [-1, 1, 1, 1, 1, 1, -1],
                     [-1, 1, 1, 1, 1, 1, -1],
                     [-1, 1, 1, 1, 1, 1, -1],
                     [-1, -1, -1, -1, -1, -1, -1],
                     [-1, -1, -1, -1, -1, -1, -1]]
-        assert_array_almost_equal(out, expected)
+        assert_array_almost_equal(out, xp.asarray(expected))
 
-    def test_watershed_ift08(self):
+    @skip_xp_backends("cupy", "pytorch", reasons=["torch.uint16"])
+    def test_watershed_ift08(self, xp):
         # Test cost larger than uint8. See gh-10069.
-        data = np.array([[256, 0],
-                         [0, 0]], np.uint16)
-        markers = np.array([[1, 0],
-                            [0, 0]], np.int8)
+        data = xp.asarray([[256, 0],
+                           [0, 0]], dtype=xp.uint16)
+        markers = xp.asarray([[1, 0],
+                              [0, 0]], dtype=xp.int8)
         out = ndimage.watershed_ift(data, markers)
         expected = [[1, 1],
                     [1, 1]]
-        assert_array_almost_equal(out, expected)
+        assert_array_almost_equal(out, xp.asarray(expected))
 
-    def test_watershed_ift09(self):
+    @skip_xp_backends("cupy", "pytorch", reasons=["torch.uint16"])
```

Comment:
this looks unintentional - the length of `reasons` should match the amount of `backends` (see https://github.com/scipy/scipy/blob/61f2a11cfdc07a7d167bd715aae6fe76d04c354e/scipy/conftest.py#L179-L205)

Reply:
The whole class is skipped under CuPy, and there is a need to skip a specific test method --- et voila, the skip machinery is what it is.
I've duplicated the cupy reason both ways for now.